src/Altinn.App.Core/Features/Payment/Models/Payer.cs

```diff
@@ -6,12 +6,12 @@ namespace Altinn.App.Core.Features.Payment.Models;
 public class Payer
 {
     /// <summary>
-    /// If the payer is a private person, this property should be set.
+    /// If the payer is a private person, this property should be set. Do not set both this and <see cref="Company"/>.
     /// </summary>
     public PayerPrivatePerson? PrivatePerson { get; set; }
 
     /// <summary>
-    /// If the payer is a company, this property should be set.
+    /// If the payer is a company, this property should be set. Do not set both this and <see cref="PrivatePerson"/>.
```

Comment:
Perhaps we should enforce this? As an example, using a validate method on the payer class:

    void Validate()
    {
        if (PrivatePerson is null && Company is null)
            throw new InvalidOperationException("Either PrivatePerson or Company must be set.");

        if (PrivatePerson is not null && Company is not null)
            throw new InvalidOperationException("Only one of PrivatePerson or Company can be set.");
    }